specification/elastic/resource-manager/Microsoft.Elastic/stable/2023-06-01/elastic.json

```diff
@@ -0,0 +1,2510 @@
+        ],
+        "responses": {
+          "200": {
+            "description": "Success"
+          },
+          "202": {
+            "description": "Accepted"
+          },
+          "default": {
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "#/definitions/ResourceProviderDefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "x-ms-examples": {
+          "createAndAssociateIPFilter_Create": {
+            "$ref": "./examples/IPTrafficFilter_Create.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Elastic/monitors/{monitorName}/createAndAssociatePLFilter": {
+      "post": {
+        "tags": [
+          "TrafficFilter"
+        ],
+        "summary": "Create and Associate private link traffic filter for the given deployment.",
+        "operationId": "createAndAssociatePLFilter_Create",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/MonitorNameParameter"
+          },
+          {
+            "in": "query",
+            "name": "name",
+            "description": "Name of the traffic filter",
+            "type": "string"
+          },
+          {
+            "in": "query",
+            "name": "privateEndpointGuid",
+            "description": "Guid of the private endpoint",
+            "type": "string"
+          },
+          {
+            "in": "query",
+            "name": "privateEndpointName",
+            "description": "Name of the private endpoint",
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success"
+          },
+          "202": {
+            "description": "Accepted"
+          },
+          "default": {
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "#/definitions/ResourceProviderDefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "x-ms-examples": {
+          "createAndAssociatePLFilter_Create": {
+            "$ref": "./examples/PrivateLinkTrafficFilters_Create.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Elastic/monitors/{monitorName}/associateTrafficFilter": {
```

Comment:
>associateTrafficFilter [](http://example.com/codeflow?start=123&length=22)

Instead of creating POST Action for creating and getting Traffic Filters (and for all such new POST Actions), how about adding these as child proxy resource types?

Reply:
@visingla-ms 
As discussed offline, we're just creating a stable version for the preview API versions which are already available and no new changes are being pushed here.
PFB the previous PRs:
https://github.com/Azure/azure-rest-api-specs/pull/22638
https://github.com/Azure/azure-rest-api-specs/pull/23343
https://github.com/Azure/azure-rest-api-specs/pull/23602
https://github.com/Azure/azure-rest-api-specs/pull/23650
https://github.com/Azure/azure-rest-api-specs/pull/24031